xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -782,12 +786,35 @@ AVDictionary* CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput()
         }
         headers.append(it->first).append(": ").append(value).append("\r\n");
       }
-      // we don't add blindly all options to headers anymore
-      // if anybody wants to pass options to ffmpeg, explicitly prefix those
-      // to be identified here
+      // here are the headers supported by ffmpeg that we ALLOW to be passed through
+      else if (name == "reconnect_at_eof" || name == "reconnect_streamed" ||
+               name == "reconnect_delay_max" || name == "icy" || name == "icy_metadata_headers" ||
+               name == "icy_metadata_packet")
+      {
+        CLog::Log(LOGDEBUG, "CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput() adding custom header option '%s: %s'", it->first.c_str(), value.c_str());
+        headers.append(it->first).append(": ").append(value).append("\r\n");
+      }
+      // here are the headers supported by ffmpeg that we DON'T ALLOW to be passed through
+      // specifically we should never pass 'http_proxy' as it would be a security risk
```

Comment:
this block doesn't make much sense in current state. if a header is not explicitly allowed, it is not forwarded.

Reply:
The idea was to be explicit on what will not be sent to ffmpeg and log it so the user knows it’s not possible. 

But I see you point, there is no functional reason to have the block.